cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/SourceControlManagementHttpHandler.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.gateway.handlers;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.BadRequestException;
+import io.cdap.cdap.common.RepositoryNotFoundException;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.common.security.AuditDetail;
+import io.cdap.cdap.common.security.AuditPolicy;
+import io.cdap.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import io.cdap.cdap.internal.app.services.SourceControlManagementService;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfigRequest;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * {@link io.cdap.http.HttpHandler} for source control management.
+ */
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}/repository")
+public class SourceControlManagementHttpHandler extends AbstractAppFabricHttpHandler {
+  private final SourceControlManagementService sourceControlService;
+  private static final Gson GSON = new Gson();
+
+  @Inject
+  SourceControlManagementHttpHandler(SourceControlManagementService sourceControlService) {
+    this.sourceControlService = sourceControlService;
+  }
+
+  @PUT
+  @Path("/")
+  @AuditPolicy(AuditDetail.REQUEST_BODY)
+  public void setRepository(FullHttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId) throws Exception {
+    RepositoryConfigRequest repoRequest = getRepository(request);
+    if (!repoRequest.shouldValidate()) {
+      if (!repoRequest.isValid()) {
```

Comment:
try to avoid inner nesting like this when possible. Normally this type of thing can be done by putting all the checks at the start of the method:

```
NamespaceId namespace = validateNamespaceId();
RepositoryConfigRequest repoRequest = getRepository(request);
if (!repoRequest.isValid()) {
  throw new BadRequestException(...);
} 

if (repoRequest.shouldValidate()) {
  // TODO: implement
  responder.setString(HttpResponseStatus.OK, ...);
  return;
}

sourceControlService.setRepository(...);
```